### PR TITLE
Remove unused LANGUAGE pragmas

### PR DESCRIPTION
**Pull request by philderbeast**
See #9110, remove unused LANGUAGE pragmas.

I missed removing some LANGUAGE pragmas after moving code between modules. A reviewer noticed but this is something HLint can check for, https://github.com/haskell/cabal/pull/10554#discussion_r1879152614.

Changed a couple of local variable names to avoid `-Wname-shadowing` (@sheaf these are `actionPtr` and `depsRes`) and added `-XDisambiguateRecordFields` in one place as HLint suggested to avoid `[GHC-76037]` when removing `-XRecordWildCards`.

To avoid `-Xname-shadowing` for the `includes` binding in `Cabal/src/Distribution/Backpack/Configure.hs`, I added an import list to the import of `Distribution.PackageDescription`.

To avoid `-Xname-shadowing` in `cabal-install/src/Distribution/Client/Errors.hs`, I imported a couple of modules qualified:

``` haskell
import qualified Distribution.Client.Types.Repo as Repo
import qualified Distribution.Client.Types.RepoName as RepoName
```

---

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).
* [ ] Is this a PR that fixes CI?

### CODE REVIEW

**Review by ffaf1 — APPROVED** [review]
Excellent

File: Cabal/src/Distribution/Simple/SetupHooks/Rule.hs [hunk 1]
```diff
@@ -612,9 +610,9 @@ mkCommand
   -> StaticPtr (arg -> res)
   -> arg
   -> Command arg res
-mkCommand dict actionPtr arg =
+mkCommand dict action arg =
```
Comment (ulysses4ever) [inline Cabal/src/Distribution/Simple/SetupHooks/Rule.hs:613]: Unrelated change?

File: cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs [hunk 2]
```diff
@@ -1,6 +1,6 @@
+{-# LANGUAGE DuplicateRecordFields #-}
```
Comment (ulysses4ever) [inline cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs:1]: Mm, how come it adds extensions?

File: Cabal/src/Distribution/Simple/SetupHooks/Rule.hs (hunk 1 above)
Comment (philderbeast) [reply Cabal/src/Distribution/Simple/SetupHooks/Rule.hs:613]: The commit with this change is "Avoid -Wname-shadowing".

File: Cabal/src/Distribution/Simple/SetupHooks/Rule.hs [hunk 3]
```diff
@@ -824,9 +822,9 @@ runRuleDynDepsCmd = \case
     }
       | Dict <- deRefStaticPtr instsPtr ->
           Just $ do
-            (deps, depsRes) <- runCommand depsCmd
+            (deps, dynDeps) <- runCommand depsCmd
```
[the remark this answers is not in the document]
Comment (philderbeast) [reply Cabal/src/Distribution/Simple/SetupHooks/Rule.hs:825]: The commit with this change is "Avoid -Wname-shadowing".

File: cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs (hunk 2 above)
Comment (philderbeast) [reply cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs]: As per the advice of HLint, `-XDisambiguateRecordFields` was needed (I'd selected `-XDuplicateRecordFields` by mistake but [this implies](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/duplicate_record_fields.html#extension-DuplicateRecordFields) `-XDisambiguateRecordFields`):

```
$ hlint cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs
cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs:3:1-32: 
  Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE RecordWildCards #-}
Perhaps you should remove it.
Note: may require `{-# LANGUAGE DisambiguateRecordFields #-}` adding to the top of the file

1 hint
```

Without it we get `[GHC-76037]` errors:

``` diff
$ git diff
diff --git \
  a/cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs \
  b/cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs
index 4f80be0f2..40d10d859 100644
--- a/cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs
+++ b/cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs
@@ -1,4 +1,3 @@
-{-# LANGUAGE DuplicateRecordFields #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE ScopedTypeVariables #-}
```

```
$ cabal repl cabal-install:unit-tests
...
tests/UnitTests/Distribution/Solver/Modular/DSL.hs:754:7: error: [GHC-76037]
    Not in scope: ‘pkgName’
    Suggested fix:
      Perhaps use record field of PackageIdentifier ‘C.pkgName’ (imported from Distribution.PackageDescription)
    |
754 |     { pkgName = C.mkPackageName (exAvName ex)
    |       ^^^^^^^

tests/UnitTests/Distribution/Solver/Modular/DSL.hs:755:7: error: [GHC-76037]
    Not in scope: ‘pkgVersion’
    Suggested fix:
      Perhaps use one of these:
        record field of PackageIdentifier ‘C.pkgVersion’ (imported from Distribution.PackageDescription),
        record field of PackageDescription ‘C.specVersion’ (imported from Distribution.PackageDescription)
    |
755 |     , pkgVersion = C.mkVersion [exAvVersion ex, 0, 0]
    |       ^^^^^^^^^^

tests/UnitTests/Distribution/Solver/Modular/DSL.hs:769:7: error: [GHC-76037]
    Not in scope: ‘pkgName’
    Suggested fix:
      Perhaps use record field of PackageIdentifier ‘C.pkgName’ (imported from Distribution.PackageDescription)
    |
769 |     { pkgName = C.mkPackageName (exInstName ex)
    |       ^^^^^^^

tests/UnitTests/Distribution/Solver/Modular/DSL.hs:770:7: error: [GHC-76037]
    Not in scope: ‘pkgVersion’
    Suggested fix:
      Perhaps use one of these:
        record field of PackageIdentifier ‘C.pkgVersion’ (imported from Distribution.PackageDescription),
        record field of PackageDescription ‘C.specVersion’ (imported from Distribution.PackageDescription)
    |
770 |     , pkgVersion = C.mkVersion [exInstVersion ex, 0, 0]
    |       ^^^^^^^^^^
...
Failed, 30 modules loaded.
```

Comment (philderbeast) [reply cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs]: I fixed the typo (HLS editor selection fumble). The added extension is now `-XDisambiguateRecordFields`.

File: Cabal/src/Distribution/Simple/SetupHooks/Rule.hs (hunk 1 above)
Comment (philderbeast) [reply Cabal/src/Distribution/Simple/SetupHooks/Rule.hs:613]: I've added this to the description. I dropped the `Ptr` suffix from one rename and looked up the `Note [Hooks Binary instances]` to pick the other name, `dynDeps`. Thanks for leaving the note, @sheaf.

https://github.com/haskell/cabal/blob/949464d333b573efb47bd84c679d4c86b21526c9/Cabal/src/Distribution/Simple/SetupHooks/Rule.hs#L729

Comment (ulysses4ever) [reply Cabal/src/Distribution/Simple/SetupHooks/Rule.hs:613]: Assume, thank you!

**Comment by philderbeast** [comment]
Label merge+no rebase is necessary when the pull request is from an organisation.
